scaffoldmaker/utils/eftfactory_tricubichermite.py

```diff
@@ -173,6 +173,354 @@ def createEftShellApexTop(self, nodeScaleFactorOffset0, nodeScaleFactorOffset1):
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D_DS3, [ (Node.VALUE_LABEL_D_DS1, [so + 1]), (Node.VALUE_LABEL_D_DS3, [so + 2]) ])
+            # 2 terms for cross derivative 1 2 to correct circular apex: -sin(theta).phi, cos(theta).phi
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS3, [ (Node.VALUE_LABEL_D_DS1, [1, so + 2, so + 3]), (Node.VALUE_LABEL_D_DS3, [so + 1, so + 3]) ])
+            # zero other cross derivative parameters
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS2, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS2DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+            ln = layer*2 + 2
+            # 2 terms for d/dxi3 via general linear map:
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D_DS3, [ (Node.VALUE_LABEL_D_DS1, so + 4), (Node.VALUE_LABEL_D_DS3, so + 5) ])
+            # 2 terms for cross derivative 1 2 to correct circular apex: -sin(theta).phi, cos(theta).phi
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS3, [ (Node.VALUE_LABEL_D_DS1, [1, so + 5, so + 6]), (Node.VALUE_LABEL_D_DS3, [so + 4, so + 6]) ])
+            # zero other cross derivative parameters
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS2, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS2DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [ 1, 1, 2, 2, 3, 4, 5, 6 ]
+        remapEftLocalNodes(eft, 6, ln_map)
+
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftWedgeRadial:  Failed to validate eft'
+        return eft
+
+    def createEftTetrahedronBottom(self, nodeScaleFactorOffset0, nodeScaleFactorOffset1, nodeScaleFactorOffsetUp):
+        '''
+        Create a tricubic hermite element field for a solid tetrahedron for the axis of a
+        solid sphere pole, with xi1 collapsed on xi3 = 0, xi2 collapsed on xi3 = 1 and 
+        xi3 radial. Each collapsed node has 3 scale factors giving the cos, sin coefficients
+        of the radial line from global derivatives, plus the arc subtended by
+        the element in radians, so the circumferential direction is rounded. Collapsed
+        node on xi2 = 0 has additional 2 scale factors giving the cos, sin coefficients of the
+        angle of inclination from global z-axis. Need to create a new template for each sector
+        around axis giving common nodeScaleFactorOffset values on common faces.
+        Suggestion is to start at 0 and add 100 for each radial line around axis.
+        :param nodeScaleFactorOffset0: offset of node scale factors at axis on xi1=0
+        :param nodeScaleFactorOffset1: offset of node scale factors at axis on xi1=1
+        :param nodeScaleFactorOffsetUp: offset of node scale factors as inclination increases
```

Comment:
offset of first scale factor for inclination at pole, increasing by 2 with each layer away from axis. Suggest starting at 100000 on axis of sphere.